include/boost/random/detail/seed_impl.hpp

```diff
@@ -281,7 +279,7 @@ void seed_array_int_impl(SeedSeq& seq, UIntType (&x)[n])
 template<int w, std::size_t n, class SeedSeq, class IntType>
 inline void seed_array_int_impl(SeedSeq& seq, IntType (&x)[n], boost::mpl::true_)
 {
-    typedef typename boost::make_unsigned<IntType>::type unsigned_array[n];
+    typedef typename boost::random::traits::make_unsigned<IntType>::type unsigned_array[n];
     seed_array_int_impl<w>(seq, reinterpret_cast<unsigned_array&>(x));
```

Comment:
We can get away with reinterpret_cast for builtin signed/unsigned types.  Not so much for user-defined types.